subsys/net/lib/lwm2m/lwm2m_engine.c

```diff
@@ -67,6 +67,8 @@ LOG_MODULE_REGISTER(LOG_MODULE_NAME);
 #define	COAP_OPTION_BUF_LEN	13
 #endif
 
+#define BINDING_OPT_MAX_LEN	3 /* "UQ" */
+#define QUEUE_OPT_MAX_LEN	2 /* "Q" */
```

Comment:
I suggest that we leave the length calculation to the compiler so better would be
```
#define BINDING_OPT_MAX_LEN	sizeof("UQ")
#define QUEUE_OPT_MAX_LEN	sizeof("Q")
```

Reply:
I did that initially, but it leads to compiler warning,
```
/home/seta/src/ncs/zephyr/subsys/net/lib/lwm2m/lwm2m_engine.c:74:33: warning: argument to 'sizeof' in '__builtin___strncpy_chk' call is the same expression as the source; did you mean to use the size of the destination? [-Wsizeof-pointer-memaccess]
   74 | #define QUEUE_OPT_MAX_LEN sizeof("Q")
      |                                 ^
/home/seta/src/ncs/zephyr/subsys/net/lib/lwm2m/lwm2m_engine.c:2047:23: note: in expansion of macro 'QUEUE_OPT_MAX_LEN'
 2047 |   strncpy(queue, "Q", QUEUE_OPT_MAX_LEN);
      |                       ^~~~~~~~~~~~~~~~~
```

So there a compiler is trying to be clever and warn us that
`strncpy(dst, "Q", sizeof("Q")` is not looking like I check the dst size.
Therefore I turned it back to just plain numbers and left the string in the comment.